### PR TITLE
Feature/redux

### PR DESCRIPTION
**Pull request by shaqdeff**
- Add Redux Toolkit.
- Write the book's actions and reducer using ducks pattern.
- Write the categories' actions and reducer using ducks pattern.
- Configure the Redux Store.

Kindly review and provide feedback.
Regards.

### CODE REVIEW

File: src/Redux/Books/Books.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+export const REMOVE_BOOK = 'bookstore-cms/books/REMOVE_BOOK';
+export const initialState = [];
+
+/* eslint-disable default-param-last */
+function booksReducer(state = initialState, action) {
+  switch (action.type) {
+    case ADD_BOOK:
+      return [...state, action.book];
+    case REMOVE_BOOK:
+      return { ...state, items: state.filter((item) => item.id !== action.id) };
```
Comment (julie-ify) [inline src/Redux/Books/Books.js]: - [ ] Your initial state is an array and therefore we need to remove the given book from the array and return the state as an array. hint;

```suggestion
    case REMOVE_BOOK:
      return state.filter(book => book.id !== id)};
```

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @shaqdeff,

Good job so far!

## Highlights!
- GitFlow ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️
- Good code organization and reducers were setup properly ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/Redux/Books/Books.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+export const ADD_BOOK = 'bookstore-cms/books/ADD_BOOK';
+export const REMOVE_BOOK = 'bookstore-cms/books/REMOVE_BOOK';
+export const initialState = [];
+
+/* eslint-disable default-param-last */
```
Comment (Deepakdanger) [inline src/Redux/Books/Books.js:5]: - [ ] Kindly remove this line  & solve the linter manually. As this is going to be mandatory in the next milestone.

File: src/Redux/Categories/Categories.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+const CHECK_STATUS = 'bookstore-cms/categories/CHECK_STATUS';
+const initialState = [];
+
+/* eslint-disable default-param-last */
```
Comment (Deepakdanger) [inline src/Redux/Categories/Categories.js:4]: - [ ] Kindly remove this line  & solve the linter manually. As this is going to be mandatory in the next milestone.

File: src/Redux/configureStore.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import { combineReducers } from 'redux';
+import { configureStore } from '@reduxjs/toolkit';
+/* eslint-disable import/extensions */
```
Comment (Deepakdanger) [inline src/Redux/configureStore.js:3]: - [ ] Kindly remove this line  & solve the linter manually. As this is going to be mandatory in the next milestone.

File: src/components/AddBook.jsx [hunk 5]
```diff
@@ -1,14 +1,37 @@
+import React, { useState } from 'react';
+import { useDispatch } from 'react-redux';
+/* eslint-disable import/extensions */
```
Comment (Deepakdanger) [inline src/components/AddBook.jsx:3]: - [ ] Kindly remove this line  & solve the linter manually. As this is going to be mandatory in the next milestone.

**Review by Deepakdanger — APPROVED** [review]
Hi @shaqdeff, 

### Status: Approved ✔️

Your project is complete!  There is nothing else to say other than... it's time to merge it  🚢 
Congratulations! 🎉 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
-  _Check the comment below._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**